### ISSUE TITLE
How to train on multi GPU

### ISSUE THREAD

**Issue by KaleidoZhouYN** [junyanz/pytorch-CycleGAN-and-pix2pix#148]
Hi:
I have trained with your code on single GPU and get a pretty nice result,excellent job!!!!

also I met something trouble when i want to accelerate with Multi GPU:

![multigpu](https://user-images.githubusercontent.com/31400191/32641309-179eeb60-c608-11e7-9a06-5ec2e038e726.png)

And the source code with multiGPU makes me confused:
options/base_options.py:        if len(self.opt.gpu_ids) > 0:
options/base_options.py:            torch.cuda.set_device(self.opt.gpu_ids[0])

Any advice?Thank you very much

**Comment by junyanz** [comment]
You can set the gpu_ids as 0,1,2. len(self.opt.gpu_ids) > 0 
checks if there are more than one gpu ids. 
torch.cuda.set_device(self.opt.gpu_ids[0]) sets the main device as the first gpu. 
See [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/137) for more discussion. The multi-GPU code should work with instancenorm.  We haven't implemented the synchronized batchnorm as mentioned [here](http://hangzh.com/SynchronizeBN/)

**Comment by KaleidoZhouYN** [comment]
@junyanz  Yes,I append the args --gpu_ids=0,1,2 and did not change the type of "norm".
will there be some trouble with pytorch version or something else?

**Comment by junyanz** [comment]
it should be fine. You can see which norm you are using, from the output log.
